### PR TITLE
fix: fix edge case on space access for member users with group access

### PR DESCRIPTION
**Pull request by rephus**
Closes:  https://github.com/lightdash/lightdash/issues/12179

### Description:

I'm just trying to patch the edge case here, I'm not super happy with this solution. But I can't think of a better solution right now. 

Thread: https://lightdash.slack.com/archives/C03MCQ08UKS/p1730368821963139?thread_ts=1730367778.344239&cid=C03MCQ08UKS

Before:

Screenshot from 2024-10-31 16-29-19

After:
Screenshot from 2024-10-31 16-52-49

Screenshot from 2024-10-31 16-58-31

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/common/src/utils/projectMemberRole.ts [hunk 1]
```diff
@@ -110,3 +110,25 @@ export const convertProjectRoleToSpaceRole = (
...
+): OrganizationMemberRole => {
+    switch (projectRole) {
+        case ProjectMemberRole.VIEWER:
+            return OrganizationMemberRole.VIEWER;
+        case ProjectMemberRole.INTERACTIVE_VIEWER:
+            return OrganizationMemberRole.VIEWER;
+        case ProjectMemberRole.EDITOR:
+            return OrganizationMemberRole.EDITOR;
+        case ProjectMemberRole.DEVELOPER:
+            return OrganizationMemberRole.EDITOR;
```
Comment (IrakliJani) [inline packages/common/src/utils/projectMemberRole.ts:125]: there is a developer role on an organization level

File: packages/common/src/utils/projectMemberRole.ts [hunk 2]
```diff
@@ -110,3 +110,25 @@ export const convertProjectRoleToSpaceRole = (
...
 };
+
+export const convertProjectRoleToOrganizationRole = (
+    projectRole: ProjectMemberRole,
+): OrganizationMemberRole => {
+    switch (projectRole) {
+        case ProjectMemberRole.VIEWER:
+            return OrganizationMemberRole.VIEWER;
+        case ProjectMemberRole.INTERACTIVE_VIEWER:
+            return OrganizationMemberRole.VIEWER;
```
Comment (IrakliJani) [inline packages/common/src/utils/projectMemberRole.ts:121]: same with interactive viewer

File: packages/common/src/utils/projectMemberRole.ts (hunk 1 above)
Comment (rephus) [reply packages/common/src/utils/projectMemberRole.ts:125]: uhh, thanks

Comment (rephus) [reply packages/common/src/utils/projectMemberRole.ts:125]: I copied this from `convertProjectRoleToSpaceRole`  and I didn't even look into the `cases` :see_no_evil:

File: packages/common/src/utils/projectMemberRole.ts [hunk 3]
```diff
@@ -110,3 +110,25 @@ export const convertProjectRoleToSpaceRole = (
...
+            return OrganizationMemberRole.DEVELOPER;
+        case ProjectMemberRole.ADMIN:
+            return OrganizationMemberRole.ADMIN;
+        default:
+            return assertUnreachable(
+                projectRole,
+                `Project role ${projectRole} does not match Organization roles`,
+            );
+    }
+};
```
Comment (IrakliJani) [inline packages/common/src/utils/projectMemberRole.ts:134]: 🤌

**Review by IrakliJani — APPROVED** [review]

**Pull request merged**